src/curvature/functions.jl

```diff
@@ -102,14 +102,14 @@ end
 Compute the full empirical Fisher.
 """
 function full(curvature::EmpiricalFisher, d::Tuple)
-    x, y = d
+    x, y = d                                                                 # where x contains the observation's features and y represents the target value
 
     loss = curvature.factor * curvature.loss_fun(x, y)
     𝐠 = gradients(curvature, x, y)
-    𝐠 = reduce(vcat, [vec(𝐠[θ]) for θ in curvature.params])
+    𝐠 = reduce(vcat, [vec(𝐠[θ]) for θ in curvature.params])                    # concatenates the gradients into a vector
```

Comment:
Theta here is an index, right? If so, I think `i` would be a better var name

Reply:
Agree.